jupyterhub/apihandlers/base.py

```diff
@@ -62,6 +65,42 @@ def check_referer(self):
             return False
         return True
 
+    @functools.lru_cache
+    def get_scope_filter(self, req_scope):
+        """Produce a filter for `*ListAPIHandlers* so that GET method knows which models to return.
+        Filter is a callable that takes a resource name and outputs true or false"""
+
+        try:
+            kind = re.search(scopes.kind_regex, req_scope).group(1)
```

Comment:
I thought we were going to move `kind` to inside `has_access` based on `isinstance(orm_resource, ...)`? Then the regex can be dropped.

Reply:
Agree. I'll remove the regex and add kind as a function argument